code/acting/src/acting/trajectory_interpolation.py

```diff
@@ -0,0 +1,132 @@
+        lin_points.append(
+            (start[0] + step_vector[0] * i,
+             start[1] + step_vector[1] * i)
+        )
+
+    return lin_points
+
+
+'''
+def circular_interpolation(start: Tuple[float, float],
+                           end: Tuple[float, float],
+                           arc_radius: float) -> List[Tuple[float, float]]:
+    """Interpolate points between start / end point
+    on top of the circular arc given by the arc radius."""
+
+    step_size = 2.0
+    sign = -1 if arc_radius < 0 else 1
+    arc_radius = abs(arc_radius)
+
+    # determine the circular angle of the arc
+    angle = asin((euclid_dist(start, end) / 2) / arc_radius) * 2
+
+    # construct the mid-perpendicular of |start, end| to determine the center
+    conn_middle = ((start[0] + end[0]) / 2, (start[1] + end[1]) / 2)
+    center_off = sqrt(pow(arc_radius, 2) - pow(euclid_dist(start, end) / 2, 2))
+    mid_perpend = rotate_vector(points_to_vector(start, end), pi/2 * sign)
+    circle_center = add_vector(conn_middle,
+                              scale_vector(mid_perpend, center_offset)
+                              )
+
+    # partition the arc into steps (-> interpol. geometries)
+    arc_circumference = arc_radius * angle        # (r * 2 pi) * (angle / 2 pi)
+    num_steps = int(arc_circumference / step_size) + 1  # each step < step size
+
+    # compute the interpolated points on the circle arc
+    vec = points_to_vector(circle_center, start)
+    rot_angles = [angle * (i / num_steps) for i in range(num_steps+1)]
+    points = [add_vector(
+                        circle_center,
+                        rotate_vector(vec, rot * sign)
+                        )
+                        for rot in rot_angles]
+
+    return points
+'''
+
+
+def _clean_route_duplicates(route: List[Tuple[float, float]],
+                            min_dist: float) -> List[Tuple[float, float]]:
+    """
+    Remove duplicates in the given List of tuples, if the distance between them
+    is less than min_dist.
+    :param route: list of points that should be cleaned up
+    :param min_dist: minimal allowed distance between points
+    :return: cleaned list of points
+    """
+    cleaned_route = [route[0]]
+    for next_p in route[1:]:
+        if euclid_dist(cleaned_route[-1], next_p) >= min_dist:
+            cleaned_route.append(next_p)
+    return cleaned_route
+
+
+def interpolate_route(orig_route: List[Tuple[float, float]], interval_m=0.5):
+    """Interpolate the given route with points inbetween,
+    holding the specified distance interval threshold."""
+
+    orig_route = _clean_route_duplicates(orig_route, 0.01)
+    route = []
+    for index in range(len(orig_route) - 1):
+        waypoints = linear_interpolation(orig_route[index],
+                                         orig_route[index + 1], interval_m)
+        route.extend(waypoints)
+
+    route = route + [orig_route[-1]]
+    return _clean_route_duplicates(route, 0.1)
```

Comment:
Did you experience a major difference when not cleaning the route again? Wouldn't it be nicer, if the linear_interpolation would only create points that don't have to be cleaned up again?

Reply:
You're right, at the moment the second call of clean route is useless. Might be useful later, but for now it will be removed as it nothing is changed.

Reply:
thanks, fixed in 03021b8